enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+
+    - Option 1 - For each namespace we will create DataVolume (CDI CRD) with RHCOS image which will be cloned for
+    each VM, masters and workers.
+    - Option 2 - We will use URL and each VM will pull the image for itself without the need for cloning.
+ 
+- Network
+
+    ####Option 1 - Pods network
+    - set cluster baseDomain as svc.cluster.local so services that we will create
+    as LoadBalancer will have the expected FQDN <service-name>.<namespace>.svc.cluster.local
+    - Create VMs with one interface on pods network.
+    - Create headless services for each VM to create DNS records for internal communication
+    between the nodes.
+    - Create services for 'api' and 'api-int' as loadbalancers between the masters
+    with MCS port (26223) and API server port (6443)
+    - Set ingress domain name for default router as subdomain of the ingress domain
+    of the infra OCP.
+    - Create in the underlying OCP route for each route in the provisioned cluster with
+    hostname value as the route on the provisioned cluster.\
+    Alternatively, if the infra OCP support wildcard route then one route from
+    type subdomain can be defined for all the routes to the provisioned cluster.
+    - Isolation will be achieved by creation of network policies that will allow traffic
+    only between VMs(pods) that are related to the same provisioned cluster.
+    
+    ####Option 2 - Secondary network (Multus)
+    - Create VMs attached to the secondary network (NAD) that was configured.
+    - Isolation achieved by the secondary network, it's up to the admin to decide how to create the
+    secondary networks that can use different VLAN/VXLAN/etc. 
+    
+- Storage
+
+    The VMs boot volumes will be PVs allocated from the infra cluster
+    
+    For PVs requested for pods running on the tenant cluster we have a few options:
+
+    ####Option 1 - Direct storage CSI 
+   The provisioned cluster will use CSI to attach storage using network to the VM guests.
+   This can be OCS CSI driver to consume storage from OCS installed on the infra
+   OpenShift as a tenant of OCS or any other external storage.
+   
+   ####Option 2 - KubeVirt CSI driver
+   Develop CSI driver for KubeVirt platform.
+   
+   This provider should forward the request to the infra cluster to allocate PV
+   from the infra cluster storageClass and attach it to the relevant VM where the PV will be exposed to the guest
+   as block device that the driver will attach to the requested pods.
+
+- Anti-affinity
+
+   The VMs will be scheduled with anti-affinity rules between the masters and between the workers in a way
+   that we will strive to spread the masters between the infra cluster nodes and same for the workers
+   to reduce the risk that outage of one worker in the infra cluster will cause major failure to a tenant.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
```

Comment:
upgrades of different versions of OCP will be interesting. How far will we allow versions to drift?  i.e. could we support an OCP 4.2 tenants cluster alongside an OCP 4.10 cluster, all hosted on an OCP 4.8 cluster?

Reply:
I don't see a reason why not, we are not relying on any feature/resource that doesn't exist in any OCP 4.x version